### ISSUE TITLE
ONNX is broken when built with ONNX_USE_LITE_PROTO=OFF: File already exists in database: onnx/onnx-ml.proto

### ISSUE THREAD

**Issue by yurivict** [onnx/onnx#6094]
Failures are like:
1. https://github.com/onnx/optimizer/issues/38
2. https://github.com/pyg-team/pytorch_geometric/issues/9220

This appears tot be a long-standing problem, at least since 2021.

I just verified that building with ONNX_USE_LITE_PROTO=ON fixes the problem.

Could you please fix it?

Version: 1.16.0

**Comment by justinchuby** [comment]
The issue predates protobuf-24.4, so I am not sure?

**Comment by justinchuby** [comment]
I think it is due to another package linking to onnx at the same time: https://groups.google.com/g/protobuf/c/RZRCnOywdSk/m/HsOaheyFCQAJ

**Comment by yurivict** [comment]
PyTorch links to onnx, and then torch_geometric links to PyTorch. What would be the other package?

**Comment by justinchuby** [comment]
PyTorch builds with ONNX. I wonder why it would fail with torch_geometric? Does setting `BUILD_SHARED_LIBS` help?

**Comment by yurivict** [comment]
BUILD_SHARED_LIBS is in which project?

**Comment by andife** [comment]
Does the problem still exist?

**Comment by shssoichiro** [comment]
Yes, I'm still experiencing this problem in July 2025. Unfortunately it's hard for me to debug because I'm not utilizing onnx directly and am not too knowledgeable about it, I'm using [this package](https://aur.archlinux.org/packages/vapoursynth-plugin-mlrt-ort-runtime-git) which uses onnx, etc. though the full chain can be followed through the dependencies list. I am not using any custom deps outside of Arch/AUR. But whenever I have this plugin installed and attempt to run Vapoursynth, I get the error mentioned in this issue:

```
E0000 00:00:1752526068.429725  171896 descriptor_database.cc:678] File already exists in database: onnx/onnx-ml.proto
F0000 00:00:1752526068.429786  171896 descriptor.cc:2519] Check failed: GeneratedDatabase()->Add(encoded_file_descriptor, size) 
```

**Comment by andife** [comment]
@shssoichiro @yurivict still a problem? or does a new onnx version solve that problem?
